app/src/main/java/com/keylesspalace/tusky/AuthorizeFollowActivity.kt

```diff
@@ -39,30 +43,76 @@ class AuthorizeFollowActivity : BaseActivity(), Injectable, OnFollowingAccountSe
     lateinit var mastodonApi : MastodonApi
 
     lateinit var recyclerView: RecyclerView
-    lateinit var cancelButton: AppCompatButton
+    lateinit var cancelButton: Button
     lateinit var adapter: FollowingAccountListAdapter
 
     private var toFollowUsername: String? = null
 
-    private val followingAccounts: MutableList<Pair<AccountEntity, FollowingAccountListAdapter.FollowState>> =
-            emptyList<Pair<AccountEntity, FollowingAccountListAdapter.FollowState>>().toMutableList()
-
     private fun broadcast(action: String, id: String) {
         val intent = Intent(action)
         intent.putExtra("id", id)
         LocalBroadcastManager.getInstance(applicationContext).sendBroadcast(intent)
     }
 
-    override fun onFollowingAccountSelected(account: AccountEntity, position: Int) {
+    override fun onFollowingAccountSelected(authorizeFollow: AuthorizeFollow) {
+        (recyclerView.adapter as FollowingAccountListAdapter).updateAccount(authorizeFollow.accountEntity, authorizeFollow.followState, anyPendingTransaction = true)
```

Comment:
You've got a field with adapter, why do you fetch it this (unholy) way?

Reply:
What do you mean by unholy there? It's definitely awful, but I'm not sure what to change it to.

Reply:
In your class you already have field named `adapter` - no need to fetch it through recycler